### ISSUE TITLE
Restore model in tensorlayer, the valid accuracy is low ,like random number

### ISSUE THREAD

**Issue by dzf19927** [tensorlayer/tensorlayer#880]
I use attention and DenseNet for classification in tensorlayer, when the attention after transition layer, restore model is ok, but when after block, after restoring the model, the valid acc is very low like a random number, and when i use avg_pool in attention after transition layer, it is ok, but when max_pool, acc is low, the question confused me so long. i use ubuntu 14.04, and test these in tensorflow 0.12 with1.2.8 tensorlayerand tensorflow 1.6 wiht tensorlayer 1.9.0, same question.

**Comment by zsdonghao** [comment]
could you put your code here.

**Comment by dzf19927** [comment]
def TCB_avg(layer, ratio):
	with tf.name_scope("TCB_avg") :
		TCB_temporal = int(layer.outputs.get_shape()[1])
		TCB_spatio = int(layer.outputs.get_shape()[2])
		out_channel = int(layer.outputs.get_shape()[-1])
		TCB_avg_Pool = tl.layers.PoolLayer(layer,
			                               ksize = [1,TCB_temporal,TCB_spatio,TCB_spatio,1],
			                               strides = [1,TCB_temporal,TCB_spatio,TCB_spatio,1],
			                               padding = 'VALID',
			                               pool = tf.nn.avg_pool3d,
			                               name = 'TCB_avg_Pool_layer')
		TCB_avg_Flatten = tl.layers.FlattenLayer(TCB_avg_Pool,
			                                     name = 'TCB_avg_Flatten_layer')
		TCB_avg_F_Dense = tl.layers.DenseLayer(TCB_avg_Flatten,
			                                   n_units = out_channel//ratio,
			                                   act = tf.nn.relu,
			                                   name = 'TCB_avg_First_Dense_layer')
		TCB_avg_S_Dense = tl.layers.DenseLayer(TCB_avg_F_Dense,
			                                   n_units = out_channel,
			                                   act = tf.nn.sigmoid,
			                                   name = 'TCB_avg_Second_Dense_layer')
		TCB_avg_Rescale = tl.layers.ReshapeLayer(TCB_avg_S_Dense,
			                                     shape = [-1,1,1,1,out_channel],
			                                     name = 'TCB_avg_Rescale_layer')
		layer.outputs = layer.outputs * TCB_avg_Rescale.outputs
		return layer
def TCB_max(layer, ratio):
	with tf.name_scope("TCB_max") :
		TCB_temporal = int(layer.outputs.get_shape()[1])
		TCB_spatio = int(layer.outputs.get_shape()[2])
		out_channel = int(layer.outputs.get_shape()[-1])
		TCB_max_Pool = tl.layers.PoolLayer(layer,
			                               ksize = [1,TCB_temporal,TCB_spatio,TCB_spatio,1],
			                               strides = [1,TCB_temporal,TCB_spatio,TCB_spatio,1],
			                               padding = 'VALID',
			                               pool = tf.nn.max_pool3d,
			                               name = 'TCB_max_Pool_layer')
		TCB_max_Flatten = tl.layers.FlattenLayer(TCB_max_Pool,
			                                     name = 'TCB_max_Flatten_layer')
		TCB_max_F_Dense = tl.layers.DenseLayer(TCB_max_Flatten,
			                                   n_units = out_channel//ratio,
			                                   act = tf.nn.relu,
			                                   name = 'TCB_max_First_Dense_layer')
		TCB_max_S_Dense = tl.layers.DenseLayer(TCB_max_F_Dense,
			                                   n_units = out_channel,
			                                   act = tf.nn.sigmoid,
			                                   name = 'TCB_max_Second_Dense_layer')
		TCB_max_Rescale = tl.layers.ReshapeLayer(TCB_max_S_Dense,
			                                     shape = [-1,1,1,1,out_channel],
			                                     name = 'TCB_max_Rescale_layer')
		layer.outputs = layer.outputs * TCB_max_Rescale.outputs
		return layer
DenseNet3D:
with tf.variable_scope('block_1'):
	block_1 = dense_block(initial_pool, is_training, 2)                    
with tf.variable_scope('transition_1'):
	tran_1 = transition_layer(block_1, is_training)
        TCB_avg_tran_1 = TCB_avg(tran_1, 4)
with tf.variable_scope('block_2'):
	block_2 = dense_block(TCB_avg_tran_1, is_training, 3)
with tf.variable_scope('transition_2'):
	tran_2 = transition_layer(block_2, is_training)
        TCB_avg_tran_2 = TCB_avg(tran_2, 4)

 these are some code, the DenseNet3D above is ok, but when I replace TCB_avg with TCB_max, something wrong like  what i asked, and after the block is same. thank you.

**Comment by dzf19927** [comment]
`def TCB_avg(layer, ratio):
	with tf.name_scope("TCB_avg") :
		TCB_temporal = int(layer.outputs.get_shape()[2])
		TCB_spatio = int(layer.outputs.get_shape()[3])
		out_channel = int(layer.outputs.get_shape()[-1])
		TCB_avg_Pool = tl.layers.PoolLayer(layer,
			                               ksize = [1,TCB_temporal,TCB_spatio,TCB_spatio,1],
			                               strides = [1,TCB_temporal,TCB_spatio,TCB_spatio,1],
			                               padding = 'VALID',
			                               pool = tf.nn.avg_pool3d,
			                               name = 'TCB_avg_Pool_layer')
		TCB_avg_Flatten = tl.layers.FlattenLayer(TCB_avg_Pool,
			                                     name = 'TCB_avg_Flatten_layer')
		TCB_avg_F_Dense = tl.layers.DenseLayer(TCB_avg_Flatten,
			                                   n_units = out_channel//ratio,
			                                   act = tf.nn.relu,
			                                   name = 'TCB_avg_First_Dense_layer')
		TCB_avg_S_Dense = tl.layers.DenseLayer(TCB_avg_F_Dense,
			                                   n_units = out_channel,
			                                   act = tf.nn.sigmoid,
			                                   name = 'TCB_avg_Second_Dense_layer')
		TCB_avg_Rescale = tl.layers.ReshapeLayer(TCB_avg_S_Dense,
			                                     shape = [-1,1,1,1,out_channel],
			                                     name = 'TCB_avg_Rescale_layer')
		layer.outputs = layer.outputs * TCB_avg_Rescale.outputs
		return layer

def TCB_max(layer, ratio):
	with tf.name_scope("TCB_max") :
		TCB_temporal = int(layer.outputs.get_shape()[2])
		TCB_spatio = int(layer.outputs.get_shape()[3])
		out_channel = int(layer.outputs.get_shape()[-1])
		TCB_max_Pool = tl.layers.PoolLayer(layer,
			                               ksize = [1,TCB_temporal,TCB_spatio,TCB_spatio,1],
			                               strides = [1,TCB_temporal,TCB_spatio,TCB_spatio,1],
			                               padding = 'VALID',
			                               pool = tf.nn.max_pool3d,
			                               name = 'TCB_max_Pool_layer')
		TCB_max_Flatten = tl.layers.FlattenLayer(TCB_max_Pool,
			                                     name = 'TCB_max_Flatten_layer')
		TCB_max_F_Dense = tl.layers.DenseLayer(TCB_max_Flatten,
			                                   n_units = out_channel//ratio,
			                                   act = tf.nn.relu,
			                                   name = 'TCB_max_First_Dense_layer')
		TCB_max_S_Dense = tl.layers.DenseLayer(TCB_max_F_Dense,
			                                   n_units = out_channel,
			                                   act = tf.nn.sigmoid,
			                                   name = 'TCB_max_Second_Dense_layer')
		TCB_max_Rescale = tl.layers.ReshapeLayer(TCB_max_S_Dense,
			                                     shape = [-1,1,1,1,out_channel],
			                                     name = 'TCB_max_Rescale_layer')
		layer.outputs = layer.outputs * TCB_max_Rescale.outputs
		return layer
DenseNet3D:
with tf.variable_scope('block_1'):
	block_1 = dense_block(initial_pool, is_training, 2)
with tf.variable_scope('transition_1'):
	tran_1 = transition_layer(block_1, is_training)
        TCB_avg_tran_1 = TCB_avg(tran_1, 4)
with tf.variable_scope('block_2'):
	block_2 = dense_block(TCB_avg_tran_1, is_training, 3)
with tf.variable_scope('transition_2'):
        tran_2 = transition_layer(block_2, is_training)
       TCB_avg_tran_2 = TCB_avg(tran_2, 4)
`
these are some code, the DenseNet3D above is ok, but when I replace TCB_avg with TCB_max, something wrong like what i asked, and after the block is same. thank you.

**Comment by zsdonghao** [comment]
you need to replace `layer.outputs = layer.outputs * TCB_avg_Rescale.outputs` by using `ElementwiseLayer` with `tf.add`, otherwise, you can't store the whole network using npz.
Alternatively, you can use `save_ckpt` instead.

**Comment by zsdonghao** [comment]
feel free to reopen if you want to discuss.

we just release 2.0.0 version
